tests/unit/src/test/scala/tests/RangeFormattingWhenSelectingSuite.scala

```diff
@@ -0,0 +1,164 @@
+package tests
+
+import munit.Location
+
+class RangeFormattingWhenSelectingSuite
+    extends BaseLspSuite("rangeFormatting") {
+  check(
+    "selecting starting the line that misindent",
+    s"""
+       |object Main {
+       |  val str = '''
+       |              |first line
+       |           <<        |second line
+       |                  |third line
+       |          |fourth line'''.stripMargin>>
+       |}""".stripMargin,
+    None,
+    s"""
+       |object Main {
+       |  val str = '''
+       |              |first line
+       |                   |second line
+       |                   |third line
+       |                   |fourth line'''.stripMargin
+       |}""".stripMargin
+  )
+
+  check(
+    "selecting starting the previous well indent line",
+    s"""
+       |object Main {
+       |  val str = '''
+       |              |first line<<
+       |                   |second line
+       |                  |third line
+       |          |fourth line'''.stripMargin>>
+       |}""".stripMargin,
+    None,
+    s"""
+       |object Main {
+       |  val str = '''
+       |              |first line
+       |              |second line
+       |              |third line
+       |              |fourth line'''.stripMargin
+       |}""".stripMargin
+  )
+
+  check(
+    "selecting starting the first line",
+    s"""
+       |object Main {
+       |  val str = '''first line<<
+       |                   |second line
+       |                  |third line
+       |          |fourth line'''.stripMargin>>
+       |}""".stripMargin,
+    None,
+    s"""
+       |object Main {
+       |  val str = '''first line
+       |              |second line
+       |              |third line
+       |              |fourth line'''.stripMargin
+       |}""".stripMargin
+  )
+
+  check(
+    "selecting starting the first line when using a pipe",
+    s"""
+       |object Main {
+       |  val str = '''|first line<<
+       |                   |second line
+       |                  |third line
+       |          |fourth line'''.stripMargin>>
+       |}""".stripMargin,
+    None,
+    s"""
+       |object Main {
+       |  val str = '''|first line
+       |               |second line
+       |               |third line
+       |               |fourth line'''.stripMargin
+       |}""".stripMargin
+  )
+
+  check(
+    "selecting entire multiline string", // wont work
+    s"""
+       |object Main {
+       |  <<val str = '''|first line
+       |                   |second line
+       |                  |third line
+       |          |fourth line'''.stripMargin>>
+       |}""".stripMargin,
+    None,
+    s"""
+       |object Main {
+       |  val str = '''|first line
+       |               |second line
+       |               |third line
+       |               |fourth line'''.stripMargin
+       |}""".stripMargin
+  )
+
+  check(
+    "selecting two string",
+    s"""
+       |object Main {
+       |<<  val firstString = '''
```

Comment:
This is an interesting case, I think it's fine to leave it as such and document it here.